### PR TITLE
first cut at geyser plug-in api

### PR DESCRIPTION
**Pull request by asiegel-jt**

### CODE REVIEW

File: src/disco/geyser/fd_geyser.h [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+   interrogate the validator state efficiently in real time. As
+   transactions are processed, callbacks are invoked
+   asynchronously. Note that firedancer itself does not use this
+   API. fd_geyser is meant for processes that do not wish to conform
+   to the tile architecture.
+
+   The callback pattern is:
+   acct_fun                      - as each account is updated
+   execute_fun                   = after each block is executed
+   block_fun                     = after each block is executed
```
Comment (nlgripto) [inline src/disco/geyser/fd_geyser.h:21]: what should execute_fun do?

Comment (asiegel-jt) [reply src/disco/geyser/fd_geyser.h:21]: explained below

**Comment by asiegel-jt** [comment]
> doesn't the sham template you're introducing here overlap with the existing tile framework and the links ?

i don't have another way to attach to tile links outside of the tile framework

File: src/util/scratch/fd_scratch.h [hunk 2]
```diff
@@ -17,7 +17,7 @@
    to turn on additional run-time checks. */
 
 #ifndef FD_SCRATCH_USE_HANDHOLDING
-#define FD_SCRATCH_USE_HANDHOLDING 0
+#define FD_SCRATCH_USE_HANDHOLDING 1
```
Comment (ripatel-fd) [inline src/util/scratch/fd_scratch.h:20]: Can we change this in a separate pr?

Comment (asiegel-jt) [reply src/util/scratch/fd_scratch.h]: reverted

**Pull request merged**